autogpt_platform/backend/backend/blocks/aryshare/post.py

```diff
@@ -0,0 +1,238 @@
+    random_post: Optional[bool] = Field(
+        None, description="Whether to generate random post text"
+    )
+    random_media_url: Optional[bool] = Field(
+        None, description="Whether to generate random media"
+    )
+    idempotency_key: Optional[str] = Field(None, description="Unique ID for the post")
+    notes: Optional[str] = Field(None, description="Additional notes for the post")
+
+
+class AyrsharePostOutput(BaseModel):
+    """Base output model for Ayrshare social media posts."""
+
+    status: str = Field(..., description="Status of the post")
+    id: str = Field(..., description="ID of the post")
+    refId: str = Field(..., description="Reference ID of the post")
+    profileTitle: str = Field(..., description="Title of the profile")
+    post: str = Field(..., description="The post text")
+    postIds: Optional[List[dict]] = Field(
+        None, description="IDs of the posts on each platform"
+    )
+    scheduleDate: Optional[str] = Field(None, description="Scheduled date of the post")
+    errors: Optional[List[str]] = Field(None, description="Any errors that occurred")
+
+
+class BaseAyrsharePost:
+    """Base class for Ayrshare social media posting blocks."""
+
+    def __init__(self, credentials: Optional[APIKeyCredentials] = None):
+        self.credentials = credentials or AYRSHARE_CREDENTIALS
+        self.client = AyrshareClient(
+            api_key=self.credentials.api_key.get_secret_value(),
+        )
+
+    def _create_post(
+        self,
+        input_data: AyrsharePostInput,
+        platforms: List[SocialPlatform],
+        profile_key: Optional[str] = None,
+    ) -> AyrsharePostOutput:
+        """Create a post on the specified platforms."""
+        response = self.client.create_post(
+            post=input_data.post,
+            platforms=platforms,
+            media_urls=input_data.media_urls,
+            is_video=input_data.is_video,
+            schedule_date=input_data.schedule_date,
+            first_comment=input_data.first_comment,
+            disable_comments=input_data.disable_comments,
+            shorten_links=input_data.shorten_links,
+            auto_schedule=input_data.auto_schedule,
+            auto_repost=input_data.auto_repost,
+            auto_hashtag=input_data.auto_hashtag,
+            unsplash=input_data.unsplash,
+            requires_approval=input_data.requires_approval,
+            random_post=input_data.random_post,
+            random_media_url=input_data.random_media_url,
+            idempotency_key=input_data.idempotency_key,
+            notes=input_data.notes,
+            profile_key=profile_key,
+        )
+        return AyrsharePostOutput(**response.__dict__)
+
+
+class PostToFacebook(BaseAyrsharePost):
+    """Block for posting to Facebook."""
+
+    def __init__(self, credentials: Optional[APIKeyCredentials] = None):
+        super().__init__(credentials)
+
+    def run(self, input_data: AyrsharePostInput, *, user_id: str) -> AyrsharePostOutput:
```

Comment:
This is how I think I should be loading the credential during run time. Is this the way it should be done?